src/main/java/org/opensearch/geospatial/ip2geo/common/InputFormatValidator.java

```diff
@@ -0,0 +1,47 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.opensearch.geospatial.ip2geo.common;
+
+import java.io.UnsupportedEncodingException;
+
+import org.opensearch.OpenSearchException;
+import org.opensearch.common.Strings;
+
+public class InputFormatValidator {
+    private static final int MAX_DATASOURCE_NAME_BYTES = 127;
+
+    public String validateDatasourceName(final String datasourceName) {
+        if (datasourceName == null || datasourceName.isEmpty()) {
+            return "datasource name must not be empty";
+        }
+        if (!Strings.validFileName(datasourceName)) {
+            return "datasource name must not contain the following characters " + Strings.INVALID_FILENAME_CHARS;
```

Comment:
should we use String.format or join instead of direct concatination

Reply:
ack